pkg/parquetquery/iters.go

```diff
@@ -80,17 +83,43 @@ func EqualRowNumber(upToDefinitionLevel int, a, b RowNumber) bool {
 		(a[2] == b[2] || upToDefinitionLevel < 2) &&
 		(a[3] == b[3] || upToDefinitionLevel < 3) &&
 		(a[4] == b[4] || upToDefinitionLevel < 4) &&
-		(a[5] == b[5] || upToDefinitionLevel < 5)
+		(a[5] == b[5] || upToDefinitionLevel < 5) &&
+		(a[6] == b[6] || upToDefinitionLevel < 6) &&
+		(a[7] == b[7] || upToDefinitionLevel < 7)
 }
 
-func TruncateRowNumber(definitionLevelToKeep int, t RowNumber) RowNumber {
+func truncateRowNumberSlow(definitionLevelToKeep int, t RowNumber) RowNumber {
 	n := EmptyRowNumber()
 	for i := 0; i <= definitionLevelToKeep; i++ {
 		n[i] = t[i]
 	}
 	return n
 }
 
+func TruncateRowNumber(definitionLevelToKeep int, t RowNumber) RowNumber {
+	switch definitionLevelToKeep {
+	case 0:
+		return RowNumber{t[0], -1, -1, -1, -1, -1, -1, -1}
+	case 1:
+		return RowNumber{t[0], t[1], -1, -1, -1, -1, -1, -1}
+	case 2:
+		return RowNumber{t[0], t[1], t[2], -1, -1, -1, -1, -1}
+	case 3:
+		return RowNumber{t[0], t[1], t[2], t[3], -1, -1, -1, -1}
+	case 4:
+		return RowNumber{t[0], t[1], t[2], t[3], t[4], -1, -1, -1}
+	case 5:
+		return RowNumber{t[0], t[1], t[2], t[3], t[4], t[5], -1, -1}
+	case 6:
+		return RowNumber{t[0], t[1], t[2], t[3], t[4], t[5], t[6], -1}
+	case 7:
+		return RowNumber{t[0], t[1], t[2], t[3], t[4], t[5], t[6], t[7]}
+	default:
+		level.Error(log.Logger).Log("msg", "definition level out of bound: should be [0:7] but got %d", definitionLevelToKeep)
+		panic("invalid definition level")
```

Comment:
fine with the panic, but would prefer not bringing a logging into this package. let's remove the logs and flesh out the panic message

use `panicWhenInvalidDefinitionLevel` below?

Reply:
this requires a return unless there's a better way to reuse the function?